test/e2e_test.go

```diff
@@ -0,0 +1,155 @@
+// +build e2e
+
+package test
+
+import (
+	"context"
+	"github.com/Shopify/sarama"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"testing"
+	"time"
+)
+
+const (
+	httpUrlPrefix   = "http://localhost:8080"
+	metricsEndpoint = "/metrics"
+)
+
+/* test checks for following:
+*  the presence of canary topic,
+*  Liveness of topic (messages being produced),
+*/
+func TestCanaryTopicLiveness(t *testing.T) {
+
+	log.Println("TestCanaryTopic test starts")
+	consumingHandler := NewConsumerGroupHandler();
+	timeout := time.After(time.Second * 10)
+
+	// test itself.
+	go func() {
+		config := sarama.NewConfig()
+		config.Consumer.Return.Errors = true
+		config.Consumer.Offsets.Initial =  sarama.OffsetOldest
+		ctx := context.Background()
+
+		newClusterAdmin, _ := sarama.NewClusterAdmin([]string{serviceManager.KafkaBrokerAddress}, config);
+		topicsDescriptionList, err := newClusterAdmin.DescribeTopics([]string{serviceManager.TopicTestName})
+		if err != nil {
+			t.Fatal("Problem with obtaining canary topic")
+		}
+		if len(topicsDescriptionList) != 1 || topicsDescriptionList[0].Name != serviceManager.TopicTestName {
+			t.Fatalf(" %s Topic isn't present", serviceManager.TopicTestName)
+		}
+
+		// consume single message
+		group, err := sarama.NewConsumerGroup([]string{serviceManager.KafkaBrokerAddress}, "faq-g9", config)
+		if err != nil {
+			panic(err)
+		}
+
+		// set up client for getting partition count on canary topic
+		client, _ := sarama.NewClient([]string{serviceManager.KafkaBrokerAddress},config )
+		listOfPartitions, _ := client.Partitions(serviceManager.TopicTestName)
+		var partitionsCount =  len(listOfPartitions)
```

Comment:
you can get the partition count from the `topicsDescriptionList[0].Partitions`, no need for a new client just for that.

Reply:
this list is empty despite existence of partitions, so I would rather stick with version that worked well.

Reply:
it's kind of impossible, it's used by the canary ... see `len(topicMetadata.Partitions)` used in the code for example. There should be some other problem is you see it empty.

Reply:
Basically you are right, will be done in a minute. 
TLDR
If I create Admin in the beginning of the test, `len(topicsDescriptionList[0].Partitions)` returns 0 whereas former solution 1 (what is correct in default test).  `len(topicsDescriptionList[0].Partitions)` returns correct value as well if I wait some extra time, probably just updated and create admin then. So I can stick to your recommendation, but I will also implement some sort of "waiting".